doc/SONIC_Test_Ingress_Discard_All_HLD.md

```diff
@@ -0,0 +1,816 @@
+| 14 | SRC IP is multicast address | IP|
+| 15 | SRC IP address is in class E | IP|
+| 16 | SRC IP address is not specified | IP|
+| 17 | DST IP address is not specified | IP|
+| 18 | ACL SRC IP DROP| IP|
+
+
+#### Related DUT CLI commands
+| **Command**                                                      | **Comment** |
+|------------------------------------------------------------------|-------------|
+| show interfaces counters              | Check ```RX_DRP```                     |
+| counterpoll rif enable                | Enable RIF counters                    |
+| show interface counters rif           | Check ```RX_ERR```                     |
+| aclshow -a                            | Check ```PACKETS COUNT```              |
+
+#### SAI attributes
+```SAI_PORT_STAT_IF_IN_DISCARDS``` - query number of all discards
+
+#### General test flow
+##### Each test case will use the following port types:
+- VLAN (T0)
+- LAG (T0, T1-LAG)
+- Router (T1, T1-LAG, PTF32)
+
+##### Sent packet number:
+N=5
+
+##### step #1 - Disable VMs
+Before test suite run - disable control plane traffic generation. Use "testbed-cli.sh" script with "disconnect-vms" option.
+
+##### step #2 - Execute test scenario for all available port types depends on run topology (VLAN, LAG, Router)
+
+- Select two pairs of PTF and DUT ports considering topology: PTF_PORT[1] <--->DUT_PORT[1], PTF_PORT[2] <--->DUT_PORT[2]
+- Clear counters. Use CLI command "sonic-clear counters"
+- Inject N packet into PTF_PORT[1] (N - depends on test case)
+- Check "RX_DRP" counter incremented on N
+	- If counter was not incremented on N, test fails with expected message
+- Check other counters were incremented on N based on sent packet type, sent port and expected drop reason (depends on test case)
+	- If counter was not incremented on N, test fails with expected message
+- Check the packet was dropped by sniffing packet absence on PTF_PORT[2]
+
+##### step #3 - Enable VMs
+Enable previously disabled VMs using "testbed-cli.sh" script with "connect-vms" option.
+
+#### Run test
+```
+py.test --inventory ../ansible/inventory --host-pattern [DEVICE] --module-path ../ansible/library/ --testbed [DEVICE-TOPO] --testbed_file ../ansible/testbed.csv --junit-xml=./target/test-reports/ --show-capture=no --log-cli-level debug -ra -vvvvv ingress_discard/test_ingress_discard.py
+```
+
+#### Test cases
+Each test case will be additionally validated by the loganalyzer utility.
+
+Each test case will run specific traffic to trigger specific discard reasone.
+
+Pytest fixture - "ptfadapter" is used to construct and send packets.
+
+After packet is sent using source port index, test framework waits during 5 seconds for specific packet did not appear, due to ingress packet drop, on one of the destination port indices.
+
+#### Test case #1
+##### Test objective
+
+Verify packet drops when SMAC and DMAC are equal
+
+Packet to trigger drop
+```
+###[ Ethernet ]###
+  dst = 00:00:00:00:01:02
+  src = 00:00:00:00:01:02
```

Comment:
Will the packet reach the DUT if the DST MAC isn't the DUT's DST MAC?

Reply:
Fixed.